### DISCUSSION TITLE
Invalid value: 32443: provided port is already allocated

### DISCUSSION THREAD

**Discussion by rangeshgupta-altair** [k3s-io/k3s#8237] [Q&A]
**Environmental Info:**
K3s Version: k3s version v1.27.4+k3s1 (36645e73)
HELM Version: Version:"v3.12.2"

Node(s) CPU architecture, OS, and Version: 
Linux ip-172-31-2-184 6.2.0-1009-aws #9~22.04.3-Ubuntu SMP Tue Aug  1 21:11:51 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:

**Describe the bug:**
While reinstalling the service getting the "**port is already allocated**" error. But after 15-20 mins it works.

> Service "project-services" is invalid: spec.ports[0].nodePort: Invalid value: 32443: provided port is already allocated

After deleting the service, verified that node port is not been used.  But still getting "port is already allocated".

> kubectl describe service --all-namespaces | grep -i nodeport

**Steps To Reproduce:**

- Install the project using helm command the first time
> helm upgrade --install --create-namespace --namespace project_name -f values.yaml project_name . --set-string timestamp=$timestamp
- Uninstall the project
> helm uninstall monitor -n project_name
- Immediately reinstall it using the helm install command.

**Expected behavior:**
Once the service is deleted. It should be available for reuse immediately 

**Actual behavior:**
After deleting the service I need to wait for 15-20 mins before installing the service again.

**Additional context / logs:**

**Comment by brandond** [comment]
Just because the chart has been uninstalled doesn't mean that the Kubernetes objects are done deleting. There may be finalizers present that are blocking actual deletion, pending cleanup of other resources. Use `kubectl get service -A -o yaml` to see if the service is still present, and if so, check the status and finalizers field to see why is has not been deleted yet.

**Reply by brandond** [reply]
Also note that this logic is part of core Kubernetes, it is not something custom to K3s.
https://github.com/kubernetes/kubernetes/blob/v1.27.4/pkg/registry/core/service/portallocator/allocator.go#L126-L130

**Reply by th0ger** [reply]
I get similar issue even when LoadBalancer Service is deleted according to `kubectl get service -A -o yaml`
